src/io.rs

```diff
@@ -27,13 +27,12 @@ pub enum EmbeddingFormat {
 /// This wrapper of `indicatif`'s `ProgressBar` updates progress based on the
 /// current offset within the file.
 impl FileProgress {
-    pub fn new(file: File) -> io::Result<Self> {
+    pub fn new(file: File) -> Result<Self> {
         let metadata = file.metadata()?;
         let progress = ProgressBar::new(metadata.len());
-        progress.set_style(
-            ProgressStyle::default_bar()
-                .template("{bar:30} {bytes}/{total_bytes} ETA: {eta_precise}"),
-        );
+        let style = ProgressStyle::default_bar()
+            .template("{bar:30} {bytes}/{total_bytes} ETA: {eta_precise}")?;
```

Comment:
I encountered this in SyntaxDot as well. I think these should probably be assertion-style/`expect` errors, since we don't use any user input in the template, failing is a programming error rather than a runtime error. It's a bit of a shame that there is no compile-time construct/check for this.

Reply:
Finally got around to push this! Changed both instances to `expect()`s